.github/workflows/test-coverage.yml

```
@@ -0,0 +1,45 @@
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle') }}
+          restore-keys: ${{ runner.os }}-gradle-
+
+      # Build and run tests with Gradle
+      - name: Build and Test
+        run: ./gradlew build jacocoTestReport --info
+
+      # Run SonarQube analysis
+      - name: SonarQube Analysis
+        env:
+          SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }} # For PR decoration
```

YAMLlint (1.35.1)

[warning] too few spaces before comment

(comments)